File: PB - More Exercises Python/7.0 Nested Loops - More Exercises/04. Car Number.py
start, end = int(input()), int(input()) + 1

for a in range(start, end):
    for b in range(start, end):
        for c in range(start, end):
            for d in range(start, end):
                if not a & 1 and d & 1 or a & 1 and not d & 1:
                    if a > d:
                        if (b + c) % 2 == 0:
                            print(str(a) + str(b) + str(c) + str(d), end=' ')

File: PB - More Exercises GO/7.0 Nested Loops - More Exercises/04. Car Number.go
package main

import "fmt"

func main() {
	var start, end int
	fmt.Scan(&start, &end)
	var result string

	for a := start; a <= end; a++ {
		for b := start; b <= end; b++ {
			for c := start; c <= end; c++ {
				for d := start; d <= end; d++ {
					if a%2 == 0 && d%2 != 0 || a%2 != 0 && d%2 == 0 {
						if a > d {
							if (b+c)%2 == 0 {
								result += fmt.Sprintf("%v%v%v%v ", a, b, c, d)
							}
						}
					}
				}
			}
		}
	}

	fmt.Println(result)
}

/*
name   :04. Car Number
input  :2 3
output :3222 3332
*/
